src/components/TreeList/TreeList.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import {useUniqId} from '../../hooks';
+import {
+    ListItemView,
+    getItemRenderState,
+    isKnownStructureGuard,
+    useList,
+    useListKeydown,
+} from '../useList';
+import type {ListItemId} from '../useList';
+import {block} from '../utils/cn';
+
+import {TreeListContainer} from './components/TreeListContainer/TreeListContainer';
+import type {TreeListProps, TreeListRenderContainerProps} from './types';
+
+import './TreeList.scss';
+
+const b = block('tree-list');
+
+export const TreeList = <T,>(props: TreeListProps<T>) => {
+    const {
+        id,
+        size = 'm',
+        items,
+        className,
+        expandedById,
+        disabledById,
+        activeItemId,
+        selectedById,
+        getId,
+        renderItem: propsRenderItem,
+        renderContainer = TreeListContainer,
```

Comment:
Suggestion: rename `renderContainer` to `containerSlot` and move `renderItem` to the `children`

Reply:
these are two different possibilities. For most tasks, you don't need to redefine the container